thunder/core/transforms.py

```diff
@@ -3733,8 +3733,19 @@ def ones_like(x):
     finally:
         reset_tracectx(tracectx_token)
 
+    saved_for_backward = forward_trace.output[1]
+    flat_saves, _ = tree_flatten(saved_for_backward)
+
     def backward_fn(saved_for_backward, cotangents):
+        # trace converts all saved_for_backward into proxy, we want to restore number scalars afterwards.
+        flat_saves_proxified, saves_spec = tree_flatten(saved_for_backward)
+        flat_filtered = [
+            proxified if isinstance(entry, Proxy) else entry
+            for proxified, entry in zip(flat_saves_proxified, flat_saves)
+        ]
+        saved_for_backward = tree_unflatten(flat_filtered, saves_spec)
```

Comment:
These lines do not change the function signature of the backward function and the saved numbers are still passed to the backward function from forward but silently ignored.
Luckily, there is a postprocessing step that removes saved tensors/numbers that are unused in the backward trace:
https://github.com/Lightning-AI/lightning-thunder/blob/f08088e65361484e34ec4ba105ea1e7b91cf91e9/thunder/core/transforms.py#L3791-L3796

Reply:
> and the saved numbers are still passed to the backward function from forward but silently ignored.

yeah that's intended~